main.c

```diff
@@ -672,6 +669,8 @@ void *socks5_thread(void *thread_data) {
 		free(auths);
 	if (thost)
 		free(thost);
+	if (hostname)
+		free(hostname);
```

Comment:
If we don't enter the "else" block `hostname` is not initialised and these "free" can lead to unexpected behaviour.
It is better to remove this "if" block and put the "free(hostname)" at the end of the "else" block

Reply:
Yep, makes sense. I made the changes. Thanks!

Be sure to let it know if you still see even more room for improvement :)

Reply:
I guess now it's ok! Thanks!